store/sql_store.go

```diff
@@ -611,3 +619,11 @@ func decrypt(key []byte, cryptoText string) (string, error) {
 
 	return fmt.Sprintf("%s", ciphertext), nil
 }
+
+// Interface for both gorp.DbMap and gorp.Transaction to allow code for one to be reused with the other
+type Queryable interface {
```

Comment:
Is this necessary?  Isn't Queryable all internal to Preferences?  can't you just use interface vs the unnecessary abstraction.